rust_dev_preview/s3/src/bin/s3-multipart-upload.rs

```diff
@@ -0,0 +1,163 @@
+        let rand_string: String = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(256)
+            .map(char::from)
+            .collect();
+        let return_string: String = "\n".to_string();
+        file.write_all(rand_string.as_ref())
+            .expect("Error writing to file.");
+        file.write_all(return_string.as_ref())
+            .expect("Error writing to file.");
+        if file.metadata().unwrap().len() > CHUNK_SIZE * 4 {
+            // Create a 5 chunk file.
+            break;
+        }
+    }
+
+    let path = Path::new(&key);
+    let file_size = tokio::fs::metadata(path)
+        .await
+        .expect("it exists I swear")
+        .len();
+
+    let mut chunk_count = (file_size / CHUNK_SIZE) + 1;
+    let mut size_of_last_chunk = file_size % CHUNK_SIZE;
+    if size_of_last_chunk == 0 {
+        size_of_last_chunk = CHUNK_SIZE;
+        chunk_count -= 1;
+    }
+
+    if file_size == 0 {
+        panic!("Bad file size.");
+    }
+    if chunk_count > MAX_CHUNKS {
+        panic!("Too many chunks! Try increasing your chunk size.")
+    }
+
+    let mut upload_parts: Vec<CompletedPart> = Vec::new();
+
+    for chunk_index in 0..chunk_count {
+        let this_chunk = if chunk_count - 1 == chunk_index {
+            size_of_last_chunk
+        } else {
+            CHUNK_SIZE
+        };
+        let stream = ByteStream::read_from()
+            .path(path)
+            .offset(chunk_index * CHUNK_SIZE)
+            .length(Length::Exact(this_chunk))
+            .build()
+            .await
+            .unwrap();
+        //Chunk index needs to start at 0, but part numbers start at 1.
+        let part_number = (chunk_index as i32) + 1;
+        // snippet-start:[rust.example_code.s3.upload_part]
+        let upload_part_res = client
+            .upload_part()
+            .key(&key)
+            .bucket(&bucket_name)
+            .upload_id(upload_id)
+            .body(stream)
+            .part_number(part_number)
+            .send()
+            .await?;
+        upload_parts.push(
+            CompletedPart::builder()
+                .e_tag(upload_part_res.e_tag.unwrap_or_default())
+                .part_number(part_number)
+                .build(),
+        );
+        // snippet-end:[rust.example_code.s3.upload_part]
+    }
+    // snippet-start:[rust.example_code.s3.upload_part.CompletedMultipartUpload]
+    let completed_multipart_upload: CompletedMultipartUpload = CompletedMultipartUpload::builder()
+        .set_parts(Some(upload_parts))
+        .build();
+    // snippet-end:[rust.example_code.s3.upload_part.CompletedMultipartUpload]
+
+    // snippet-start:[rust.example_code.s3.complete_multipart_upload]
+    let _complete_multipart_upload_res = match client
+        .complete_multipart_upload()
+        .bucket(&bucket_name)
+        .key(&key)
+        .multipart_upload(completed_multipart_upload)
+        .upload_id(upload_id)
+        .send()
+        .await
+    {
```

Comment:
Isn't this `.unwrap()`?

Reply:
Haha yep! I'm pretty sure when that section started, I needed the extra logic. But over time, it became pared down and I never noticed. Thanks for the catch! Fixed.